### PR TITLE
#430: Use proper plurals based on image settings

### LINKED ISSUE #430: Use proper plurals based on image settings
### Is your enhancement related to a problem? Please describe.

In #419, we added a feature that allows you to generate one or more images using the OpenAI service and then import one or more images into your media library.

Within the service settings, you can choose how many images should be generated each time you run a prompt. This can be anything from 1 to 10. Within the interface where image generation happens, there's a few places that we just assume multiple images will be generated, like the main tab that says `Generate images` or the prompt button that says `Generate images` or the descriptive text that says `Enter a prompt below to generate images` (and there's a handful of other places as well).

This is a minor thing so may be worth just closing out but we could look to change this text based on the `Number of images` setting. If that's set to one, change from using plural `images` to just using `image`.

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by jamesmorrison**
### Description of the Change
Adds singular labels where a single image is selected for generation.

Closes #430

### How to test the Change

**Test 1: Plural**:
- Open Classifai settings:
  - set the number of images to generate between 2 - 10 (i.e. not 1)
  - Check / ensure: `Enable image generation` is checked
- Create a new post:
  - Open the media modal (in a new post => featured image / media block) and select `Generate Images`
  - Confirm the text remains as:
    - `Enter a prompt below to generate images.`
    - `Once images are generated, choose one or more of those to import into your Media Library and then choose one image to insert.`

**Test 3: Singular**:
- Open Classifai settings:
  - set the number of images to generate 1 (i.e. not 2 - 10)
  - Check / ensure: `Enable image generation` is checked
- Create a new post:
  - Open the media modal (in a new post => featured image / media block) and select `Generate Images`
  - Confirm the text shows singular terminology:
    - `Enter a prompt below to generate an image.` - new
    - `Once an image is generated, you can import it into your Media Library and then select to insert.` - new

### Credits
Props @jamesmorrison

### Checklist:
- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

**Comment by jamesmorrison** [comment]
@jeffpaul / @dkotter Please could you review this change? If you'd prefer alternative singular test, please let me know what this should be updated to - I added what I thought was appropriate here.

File: includes/Classifai/Providers/OpenAI/DallE.php [hunk 1]
```diff
@@ -128,16 +128,30 @@ public function enqueue_admin_scripts( $hook_suffix = '' ) {
...
+		$number_of_images = absint( $settings['number'] );
 		?>
 
 		<?php // Template for the Generate images tab content. Includes prompt input. ?>
 		<script type="text/html" id="tmpl-dalle-prompt">
 			<div class="prompt-view">
 				<p>
-					<?php esc_html_e( 'Enter a prompt below to generate images.', 'classifai' ); ?>
+					<?php
+					if ( $number_of_images > 1 ) {
```
Comment (ravinderk) [inline includes/Classifai/Providers/OpenAI/DallE.php:152]: We can use [`_n`](https://developer.wordpress.org/reference/functions/_n/) function to handle plural translation.

File: includes/Classifai/Providers/OpenAI/DallE.php [hunk 2]
```diff
@@ -128,16 +128,30 @@ public function enqueue_admin_scripts( $hook_suffix = '' ) {
...
+						esc_html_e( 'Enter a prompt below to generate images.', 'classifai' );
+					} else {
+						esc_html_e( 'Enter a prompt below to generate an image.', 'classifai' );
+					}
+					?>
 				</p>
 				<p>
-					<?php esc_html_e( 'Once images are generated, choose one or more of those to import into your Media Library and then choose one image to insert.', 'classifai' ); ?>
+					<?php
+					if ( $number_of_images > 1 ) {
```
Comment (ravinderk) [inline includes/Classifai/Providers/OpenAI/DallE.php:161]: We use [`_n`](https://developer.wordpress.org/reference/functions/_n/) function to handle plural translation.

**Review by ravinderk — CHANGES_REQUESTED** [review]

File: includes/Classifai/Providers/OpenAI/DallE.php (hunk 1 above)
Comment (jamesmorrison) [reply includes/Classifai/Providers/OpenAI/DallE.php:152]: Thanks @ravinderk; I've updated this.

File: includes/Classifai/Providers/OpenAI/DallE.php (hunk 2 above)
Comment (jamesmorrison) [reply includes/Classifai/Providers/OpenAI/DallE.php:161]: Thanks @ravinderk; I've updated this.

File: includes/Classifai/Providers/OpenAI/DallE.php (hunk 1 above)
Comment (dkotter) [reply includes/Classifai/Providers/OpenAI/DallE.php:152]: Apologies for not seeing this sooner but I don't think this is the approach we should take. While it is cleaner, some languages use the singular form for numbers other than 1 (see https://codex.wordpress.org/I18n_for_WordPress_Developers#Plurals). I suggest we go back to the original approach [here](https://github.com/10up/classifai/pull/482/commits/d097367e5c51bfc3d16ff86a3eeed76538c632cf)

**Comment by dkotter** [comment]
@jamesmorrison Thanks for the work here! I've left one comment that I think we should address. There's also a few other text strings that we probably want to update.

- The `Generate images` tab text: https://github.com/10up/classifai/blob/develop/includes/Classifai/Providers/OpenAI/DallE.php#L126
- The `Generate images` button text: https://github.com/10up/classifai/blob/develop/includes/Classifai/Providers/OpenAI/DallE.php#L150
- Text shown after an image is generated: https://github.com/10up/classifai/blob/develop/includes/Classifai/Providers/OpenAI/DallE.php#L154

File: includes/Classifai/Providers/OpenAI/DallE.php (hunk 1 above)
Comment (jamesmorrison) [reply includes/Classifai/Providers/OpenAI/DallE.php:152]: @dkotter Ok, I've reverted this.

**Comment by jamesmorrison** [comment]
@dkotter I've reverted to if else (rather than `_n()`) and added singular text where this was missed previously. I've tested and confirmed this works; please could you check this?

**Review by dkotter — APPROVED** [review]

**Pull request merged**